### ISSUE TITLE
[ASK] How to cast a HLS file to chromecast from iOS 

### ISSUE THREAD

**Issue by fernandatoledo** [jwplayer/jwplayer-react-native#54]
**How can we cast a HLS video to chromecast running on iOS?**
Hey, I am having an issue when trying to cast some HLS videos to chromecast from the iOS app.

My app can detect chromecasts on the same wifi and connect to them, it can also cast some HLS videos but not all of them.

On android I can cast all HLS videos without issues.

You can see my sample URL here: https://player.vimeo.com/external/973551660.m3u8?s=1d7f0ee209cadbc191c6743d791dfca023ea8976&logging=false

I am able to play the video on the player but when casting to the chromecast it just locks there.

An issue that does not happen with this other video: https://player.vimeo.com/external/953289025.m3u8?s=e84b7b3d4be1380181d64500423ae4b8606982a1&logging=false

Can you please help?

<img src='https://github.com/jwplayer/jwplayer-react-native/assets/13571788/e69b5f7f-ef80-47ba-b796-d48f0e2a40e3' width=250 />

Even though it says that is playing on dining tv, it is not, the tv just shows the 'cast icon' and it's stuck there.

I've also tried playing dash files but JWPlayer says the format is incorrect:

202630 | Either an empty playlist was requested or none of the items passed our filter for valid files. | Playlist error: No playable sources found | Sorry, the video player failed to load.
-- | -- | -- | --

**Comment by Jmilham21** [comment]
@fernandatoledo this smells like bad media to me. See the below errors I got from my custom receiver when attempting to cast your media. This media fails to cast on all of our platforms (Web/Android/iOS). 

![image](https://github.com/jwplayer/jwplayer-react-native/assets/22652238/92578de5-5822-4630-a0b0-11710726779e)

**Comment by fernandatoledo** [comment]
Hey @Jmilham21 Thank you so much for your answer and info. I know this might be out of the scope of JWPlayer given the info you provided, but if the problem is the media, do you have any clue on why the casting from Android would work with the exact same video?

**Comment by Jmilham21** [comment]
https://player.vimeo.com/external/973551660.m3u8?s=1d7f0ee209cadbc191c6743d791dfca023ea8976&logging=false didn't work for me on our Android SDK, I didn't test it with React Native but if it fails on the standard SDK it shouldn't work here. 

I found a solution via this [post](https://stackoverflow.com/questions/55224639/does-the-chromecast-support-casting-videos-from-reddit-hls-and-dash-videos) on Stack Overflow. You may need to implement a custom Chromecast receiver for HLS formatted this way (AAC in MPEG-TS)

**Comment by Jmilham21** [comment]
Let us know if this works for you / if you need assistance putting your custom receiver ID into the SDKs. 

I'll leave this open for now. Awaiting your update

**Comment by fernandatoledo** [comment]
I can't believe it, I tested it on android after your message and it's not working, but they were working! So definitely must be the video data.

Thank you so much for your help!!

I managed to find a solution using the default google cast app, a prop needs to be set in the `GCKMediaInformation`

Here's an example for an objective-C cast player: 

```
    GCKMediaInformationBuilder *mediaInfoBuilder =
        [[GCKMediaInformationBuilder alloc] initWithContentURL:url];
    mediaInfoBuilder.streamDuration = duration;
    mediaInfoBuilder.streamType = GCKMediaStreamTypeBuffered;
    mediaInfoBuilder.contentType = mimeType;
    mediaInfoBuilder.metadata = metadata;
    mediaInfoBuilder.mediaTracks = mediaTracks;
    mediaInfoBuilder.textTrackStyle = _trackStyle;
    mediaInfoBuilder.hlsSegmentFormat = GCKHLSSegmentFormatTS;
```

I would love to be able to adjust the JWPlayer library so that we can set that property but it's a compiled library, is there some workaround for this?

Trying to find a workaround as quickly as possible here, we're really close to our launch date, thank you1

**Comment by Jmilham21** [comment]
@fernandatoledo I think if you need a very quick solution, you'll want to [create](https://developers.google.com/cast/docs/web_receiver/basic) and [register](https://developers.google.com/cast/docs/web_receiver/basic#register_your_web_receiver_app) a custom Chromecast receiver with the logic on the receiver side, as the sender logic lives in our Native SDK and there is no way to modify this via this wrapper. 

I see that specifying the `hlsSegmentFormat` for TS does solve this. I have created internal tickets to patch this in the native SDKs. Then, we will update this wrapper when those changes are released. 

Sorry for no quick or simple workaround. I do think a custom receiver will work if you need to launch before we can go through the process of updating both native SDKs to support streams in this format.

**Comment by fernandatoledo** [comment]
Hello @Jmilham21 we've tested it and the custom receiver app works indeed :D Thanks for the help!

**Comment by Jmilham21** [comment]
Glad to hear it @fernandatoledo. We will update the native cast implementation in a future release, and update you here when it's done and released in this wrapper.

**Comment by Jmilham21** [comment]
Hey @fernandatoledo, after some internal discussion, we have decided we won't be able to modify the native SDK to support this stream with casting out of the box. We suggest that if you require streams with TS-formatted audio segments, you continue handling the configuration of the cast receiver, as we've discussed in this thread. 

I'm sorry for the delay in this update, but we were just able to look into it more deeply. If this is a significant blocker for you, I'd suggest raising it with your company's account manager / through our support channels.
